From 88d3c5010a207c651d1940856d0d26935434e3c2 Mon Sep 17 00:00:00 2001
From: Holger Freyther <ich@tamarin.(none)>
Date: Tue, 13 May 2008 18:53:40 +0100
Subject: [PATCH] Explain why we want SECOND to kick.
     We do not mask the SECOND interrupt to make sure that we get
     a interrupt at the beginning. We use this for the coldplug_done
     initisalisation.
     So pcf->flags and INT1M is out of sync at the beginning but after the
     first SECOND interrupt it will be put into sync.

Signed-Off-By: Holger Freyther <zecke@openmoko.org>
---
 drivers/i2c/chips/pcf50633.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index 205f4e3..1c10c42 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -1755,7 +1755,8 @@ static int pcf50633_detect(struct i2c_adapter *adapter, int address, int kind)
 		goto exit_sysfs;
 
 	/* configure interrupt mask */
-	reg_write(data, PCF50633_REG_INT1M, 0x00); /* we want SECOND to kick */
+	/* we want SECOND to kick for the coldplug initialisation */
+	reg_write(data, PCF50633_REG_INT1M, 0x00);
 	reg_write(data, PCF50633_REG_INT2M, 0x00);
 	reg_write(data, PCF50633_REG_INT3M, 0x00);
 	reg_write(data, PCF50633_REG_INT4M, 0x00);
-- 
1.5.6.5

